### PR TITLE
C-V2X integration test

### PR DESCRIPTION
**Pull request by chengyuan0124**
# PR Details
## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: co-simulation/bundle/src/assembly/resources/scenarios/Tiergarten/ns3/ns3_federate_config.xml [hunk 1]
```diff
@@ -1,5 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <ns3>
+
+ <NetworkConfig>
+     <component name="CommType" value="LTE"/>
+     <component name="NumOfNodes" value="5"/>
```
Comment (paulbourelly999) [inline co-simulation/bundle/src/assembly/resources/scenarios/Tiergarten/ns3/ns3_federate_config.xml:6]: Does this determine the number of radios available or is that set somewhere else.

File: co-simulation/bundle/src/assembly/resources/scenarios/Tiergarten/ns3/ns3_federate_config.xml
Comment (paulbourelly999) [inline co-simulation/bundle/src/assembly/resources/scenarios/Tiergarten/ns3/ns3_federate_config.xml:1]: Why are we changing this for other scenarios? Have we confirmed these scenarios work with this change?

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -5,4 +5,4 @@ on:
 
 jobs:
   docker:
-    uses: usdot-fhwa-stol/actions/.github/workflows/docker.yml@main
+    uses: usdot-fhwa-stol/actions/.github/workflows/docker.yml@main
```
Comment (paulbourelly999) [inline .github/workflows/docker.yml:8]: Is anything changed here. If not can we remove these changes.

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]

File: .github/workflows/docker.yml (hunk 2 above)
Comment (chengyuan0124) [reply .github/workflows/docker.yml:8]: This change is for figuring out the merge conflict

File: co-simulation/bundle/src/assembly/resources/scenarios/Tiergarten/ns3/ns3_federate_config.xml (hunk 1 above)
Comment (chengyuan0124) [reply co-simulation/bundle/src/assembly/resources/scenarios/Tiergarten/ns3/ns3_federate_config.xml:6]: The NumOfNodes variable in ns3_federate_config.xml is essential for predefining and initializing the C-V2X node. In the context of C-V2X mode, numerous NS-3 helpers are employed to configure the NS-3 node for C-V2X communication. However, unlike DSRC mode where nodes are dynamically created, some of these helpers require a setup at time=0. This necessity has been verified by the leader of the NS-3 development team. As a result, in C-V2X mode, the provision of the NumOfNodes variable is mandatory

File: co-simulation/bundle/src/assembly/resources/scenarios/Tiergarten/ns3/ns3_federate_config.xml
Comment (chengyuan0124) [reply co-simulation/bundle/src/assembly/resources/scenarios/Tiergarten/ns3/ns3_federate_config.xml:1]: The changes, including CommType, NumOfNodes, and settings for LteUePhy, LteUePowerControl, and LteUeMac, are necessary for C-V2X mode functionality. They don't affect DSRC mode. We have tested these changes in the C-V2X integration testing phase for those scenarios without any issue. These additions ensure C-V2X simulations run correctly

**Review by paulbourelly999 — APPROVED** [review]

**Pull request merged**